apps/products-feed/src/modules/trpc/protected-client-procedure.ts

```diff
@@ -98,22 +101,36 @@ const validateClientToken = middleware(async ({ ctx, next, meta }) => {
  *
  * Can be used only if called from the frontend (react-query),
  * otherwise jwks validation will fail (if createCaller used)
- *
- * TODO Rethink middleware composition to enable safe server-side router calls
  */
 export const protectedClientProcedure = procedure
   .use(attachAppToken)
   .use(validateClientToken)
-  .use(async ({ ctx, next }) => {
-    const client = createClient(ctx.saleorApiUrl, async () =>
-      Promise.resolve({ token: ctx.appToken })
-    );
+  .use(async ({ ctx, next, path, type }) => {
+    const client = GraphqlClientFactory.fromAuthData({
+      token: ctx.appToken!,
+      saleorApiUrl: ctx.saleorApiUrl,
+    });
+
+    const metadataManager = new AppConfigMetadataManager(createSettingsManager(client));
 
     return next({
       ctx: {
         apiClient: client,
         appToken: ctx.appToken,
         saleorApiUrl: ctx.saleorApiUrl,
+        appId: ctx.appId,
+        appConfigMetadataManager: metadataManager,
+        getConfig: async () => {
+          const metadata = await metadataManager.get();
+
+          return metadata ? AppConfig.parse(metadata) : new AppConfig();
+        },
+        logger: createLogger({
+          appId: ctx.appId,
+          apiUrl: ctx.saleorApiUrl,
+          type,
+          path,
+        }),
```

Comment:
@peelar @krzysztofwolski we may extend context and hide boilerplate services here